lib/iris/cube.py

```diff
@@ -64,7 +65,8 @@ class CubeMetadata(collections.namedtuple('CubeMetadata',
                                            'var_name',
                                            'units',
                                            'attributes',
-                                           'cell_methods'])):
+                                           'cell_methods',
+                                           'fill_value'])):
```

Comment:
@marqh Didn't we want to add `shape` to the `CubeMetadata` ? The `shape` of a cube is immutable, and it takes care of a corner case failure that @djkirkham found the other day, when comparing cubes with the same data values but different broadcastable shapes and no coordinates e.g.
```python
>>> a = np.arange(5).reshape(1, 5)
>>> b = np.array([0, 1, 2, 3, 4, 0, 1, 2, 3, 4]).reshape(2, 5)
>>> ca = iris.cube.Cube(a)
>>> cb = iris.cube.Cube(b)
>>> ca == cb
True
```